src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -938,10 +958,21 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
     void TermControl::_KeyHandler(Input::KeyRoutedEventArgs const& e, const bool keyDown)
     {
+        if (e.OriginalKey() == VirtualKey::Enter)
+        {
+            return;
+        }
+
         // If the current focused element is a child element of searchbox,
         // we do not send this event up to terminal
-        if (_searchBox && _searchBox->ContainsFocus())
+        if (_searchBox && _searchBox->ContainsFocus() && _searchBox->TextBox().Text().empty())
+        {
+            return;
+        }
+
+        if (keyBindingSearch)

```

Comment:
Can you please document your logic here as well? If we searched with key binding then we should ignore the next key?

Reply:
Curious. If we use the keybinding to invoke this action, then on the keyDown, we'll select the text. Then on the key_up_, we'll dismiss the selection.

Reply:
Okay this is definitely tricky, and I don't know what a good solution is. 

The action is either invoked:
* with a keybinding, in the `TermControl::_KeyHandler(true, ...)` method
* by some other magic way (command palette)? 

So if it happened on a keydown, we do need to _not_ dismiss the selection on the next keyup. Weird. I wonder how this worked in the keyboard-selection implementation. Wouldn't that have hit the same problem? Paging @carlos-zamora to see if he can here. Am I just being daft? Is there an obvious way to create a selection via action and have it not dismiss on the keyup?

Reply:
Derp. I can't read. 
https://github.com/microsoft/terminal/pull/3758#issuecomment-658997608

We need to do this for #3758 so I'm just gonna do it here anyways